stdlib/public/Reflection/TypeRefBuilder.cpp

```diff
@@ -107,7 +107,7 @@ TypeRefBuilder::normalizeReflectionName(RemoteRef<char> reflectionName) {
       if (!mangling.isSuccess()) {
         return {};
       }
-      return mangling.result();
+      return std::move(mangling.result());
```

Comment:
Actually, I think this should be reverted, since it disables [RVO](https://stackoverflow.com/a/4986802)

Reply:
My understanding is that RVO doesn't apply here, as this value is of type `string`, but the return type is `Optional<string>`. This change does not trigger the `-Wpessimizing-move` warning.

Reply:
I'll restate that, I think RVO does happen here – for the `Optional`. This `move` applies to the value going inside of the `Optional`, not the `Optional` itself.

Reply:
maybe this would be more clear if the code was explicit:

```cpp
return Optional(std::move(…));
```

Reply:
I see, makes sense!